app/models/course/forum/topic.rb

```diff
@@ -4,12 +4,53 @@ class Course::Forum::Topic < ActiveRecord::Base
 
   acts_as :topic, class_name: Course::Discussion::Topic.name, inverse_of: :actable
 
+  before_validation :set_initial_post_title
```

Comment:
I think this should not be changed. Because the initial post can be deleted or edited. I'll add `unless: persisted?`